### PR TITLE
fix: print detailed error

### LINKED ISSUE #3138: Detail error didn't print during the procedure execution
### What type of enhancement is this?

User experience

### What does the enhancement do?

We should let the system print the detailed error info instead of the `xxx error`
image

### Implementation challenges

_No response_

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

fix missing detailed error

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

## Refer to a related PR or issue link (optional)
#2700

fixes #3138

### CODE REVIEW

File: src/meta-srv/src/procedure/region_failover/activate_region.rs [hunk 1]
```diff
@@ -162,19 +164,26 @@ impl ActivateRegion {
...
+                    let violated = format!(
                         "Region {failed_region:?} is not opened by Datanode {:?}, error: {error:?}",
                         self.candidate,
                     );
-                    RetryLaterSnafu { reason }.fail()
+
+                    UnexpectedSnafu { violated }
+                        .fail()
+                        .map_err(BoxedError::new)
+                        .context(RetryLaterSnafu)
```
Comment (MichaelScofield) [inline src/meta-srv/src/procedure/region_failover/activate_region.rs:175]: Why add an indirection layer of `UnexpectedSnafu` to throw the retry error?

Comment (WenyXu) [reply src/meta-srv/src/procedure/region_failover/activate_region.rs]: PTAL, I refactor this part.

File: src/meta-srv/src/error.rs [hunk 2]
```diff
@@ -540,6 +540,13 @@ pub enum Error {
     #[snafu(display("Expected to retry later, reason: {}", reason))]
     RetryLater { reason: String, location: Location },
 
+    #[snafu(display("Expected to retry later, reason: {}", reason))]
+    RetryLaterWithSource {
```
Comment (MichaelScofield) [inline src/meta-srv/src/error.rs:544]: Can we get rid of the old `RetryLater` and make `RetryLaterWithSource` the new `RetryLater` error? It seems all retry later errors should have sources.

Comment (WenyXu) [reply src/meta-srv/src/error.rs:544]: Not all retries have an error inside; some are just error messages. This is why I add a  indirection layer before.

**Review by MichaelScofield — APPROVED** [review]

**Review by waynexia — APPROVED** [review]
LGTM

**Pull request merged**